### PR TITLE
IH-98 Add api update post

### PR DESCRIPTION
**Pull request by Sotatek-KimMai**

### CODE REVIEW

File: src/main/java/art/heredium/service/PostService.java [hunk 1]
```diff
@@ -186,4 +196,226 @@ private String moveEditorContent(final String editorContent, final String newPat
...
+  private MembershipCouponCreateRequest convertToMembershipCouponCreateRequest(
+      MembershipCouponUpdateRequest updateRequest) {
+    MembershipCouponCreateRequest createRequest = new MembershipCouponCreateRequest();
+    createRequest.setName(updateRequest.getName());
+    createRequest.setCouponType(updateRequest.getCouponType());
+    createRequest.setDiscountPercent(updateRequest.getDiscountPercent());
+    createRequest.setPeriodInDays(updateRequest.getPeriodInDays());
+    createRequest.setImageUrl(updateRequest.getImageUrl());
+    createRequest.setNumberOfUses(updateRequest.getNumberOfUses());
+    createRequest.setIsPermanent(updateRequest.getIsPermanent());
```
Comment (Sotatek-ThoLe) [inline src/main/java/art/heredium/service/PostService.java:352]: Làm cái @Builder cho MembershipCouponCreateRequest để code dc gọn hơn.

File: src/main/java/art/heredium/service/PostService.java [hunk 2]
```diff
@@ -186,4 +196,226 @@ private String moveEditorContent(final String editorContent, final String newPat
...
+
+  private void createNewMembership(Post post, PostMembershipUpdateRequest request) {
+    MembershipCreateRequest createRequest = new MembershipCreateRequest();
+    createRequest.setName(request.getName());
+    createRequest.setPrice(request.getPrice());
+    createRequest.setImageUrl(request.getImageUrl());
+    createRequest.setCoupons(convertToMembershipCouponCreateRequests(request.getCoupons()));
+
+    List<MembershipCreateRequest> createRequests = List.of(createRequest);
+    membershipService.createMemberships(post.getId(), createRequests);
```
Comment (Sotatek-ThoLe) [inline src/main/java/art/heredium/service/PostService.java:331]:
```suggestion
    membershipService.createMemberships(post.getId(), Arrays.asList(createRequest));
```
1. List.of chỉ có từ java 9 nên ko dùng dc
2. Ko cần tạo biến mới, nên lồng luôn 2 trong 1 để code đỡ bị dài.

File: src/main/java/art/heredium/service/PostService.java [hunk 3]
```diff
@@ -186,4 +196,226 @@ private String moveEditorContent(final String editorContent, final String newPat
...
+    if (!StringUtils.isEmpty(noteImageUrl)) {
+      String fileFolderPath = String.format("%s/%d", FilePathType.POST.getPath(), post.getId());
+      validateImage(noteImageUrl);
+      String newImageUrl = this.moveImageToNewPlace(noteImageUrl, fileFolderPath);
+      post.setImageUrl(newImageUrl);
+    }
+  }
+
+  private void updateAdditionalInfo(Post post, PostUpdateRequest.AdditionalInfo additionalInfo) {
+    if (additionalInfo == null) return;
```
[the remark this answers is not in the document]
Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/service/PostService.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/68/commits/c5e4bdc0f2a7401e4ddd733dc5e5793f4c9540e0

File: src/main/java/art/heredium/service/PostService.java [hunk 4]
```diff
@@ -186,4 +196,226 @@ private String moveEditorContent(final String editorContent, final String newPat
...
+  }
+
+  private void updateMemberships(Post post, List<PostMembershipUpdateRequest> membershipRequests) {
+    if (membershipRequests == null) return;
+
+    for (PostMembershipUpdateRequest membershipRequest : membershipRequests) {
+      if (membershipRequest.getId() != null) {
+        Membership membership =
+            post.getMemberships().stream()
+                .filter(m -> m.getId().equals(membershipRequest.getId()))
```
[the remark this answers is not in the document]
Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/service/PostService.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/68/commits/c5e4bdc0f2a7401e4ddd733dc5e5793f4c9540e0

File: src/main/java/art/heredium/service/PostService.java [hunk 5]
```diff
@@ -186,4 +196,226 @@ private String moveEditorContent(final String editorContent, final String newPat
...
+                .findFirst()
+                .orElseThrow(() -> new ApiException(ErrorCode.MEMBERSHIP_NOT_FOUND));
```
[the remark this answers is not in the document]
Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/service/PostService.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/68/commits/c5e4bdc0f2a7401e4ddd733dc5e5793f4c9540e0

File: src/main/java/art/heredium/service/PostService.java (hunk 1 above)
Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/service/PostService.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/68/commits/c5e4bdc0f2a7401e4ddd733dc5e5793f4c9540e0

File: src/main/java/art/heredium/service/PostService.java (hunk 2 above)
Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/service/PostService.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/68/commits/c5e4bdc0f2a7401e4ddd733dc5e5793f4c9540e0

**Review by Sotatek-ThoLe — APPROVED** [review]

**Pull request merged**